### ISSUE TITLE
Component Accessibility Status 🏖️ 

### ISSUE THREAD

**Issue by jeanservaas** [carbon-design-system/carbon#13745]
### Summary

REF https://github.com/carbon-design-system/carbon-platform/issues/803
REF https://github.com/carbon-design-system/carbon/issues/11370

We want to give users immediate confidence that Carbon components have been vetted for accessibility out-of-the-box. Also, ideally the majority of these tests should be automated; right now they are not.

A crucial aspect of component completeness is knowing to what degree is each asset (e.g. component) accessible. After talking with Marc Johlic, we don't have to go to lengths for each asset to specify which WCAG rules they conform to (e.g. WCAG 2.1 AA.) Instead, a simpler accessibility testing status scale/process can be established:

1. **Pass** There are no violations when running the automated [Equal Access Checker](https://www.ibm.com/able/toolkit/tools) on default component state.

2. **Pass** There are no violations when running the automated [Equal Access Checker](https://www.ibm.com/able/toolkit/tools) on more complex component states and variants.

3. **Pass** There no violations when running automated keyboard navigation tests using Jess/Michaels' tabbing guidelines.

4.  **Pass** No screenreader violations. (Manual process, leverage Ragu's expertise and time.)

-------

_- if any of these show a violation it's a Sev2 bug_
_- we're ensuring that all of these steps are a pass before each new release_

-------

### Justification

We've had specific asks from leadership to offer more clarity and emphasis around the accessibility of Carbon out-of-the-box. 

### Tasks
- [ ] https://github.com/carbon-design-system/carbon/issues/13746
- [ ] https://github.com/carbon-design-system/carbon/issues/13780
- [ ] https://github.com/carbon-design-system/carbon/issues/13747
- [ ] https://github.com/carbon-design-system/carbon/issues/14480
- [ ] https://github.com/carbon-design-system/carbon/issues/13748
- [ ] https://github.com/carbon-design-system/carbon-website/issues/3520
- [ ] https://github.com/carbon-design-system/carbon/issues/14695
- [ ] https://github.com/carbon-design-system/carbon/issues/14484
- [ ] https://github.com/carbon-design-system/carbon/issues/14595
- [ ] https://github.com/carbon-design-system/carbon-website/issues/3518
- [ ] https://github.com/carbon-design-system/carbon/issues/15212
- [ ] https://github.com/carbon-design-system/carbon/issues/15326
- [ ] https://github.com/carbon-design-system/carbon/issues/15279
- [ ] https://github.com/carbon-design-system/carbon-website/issues/3848
- [ ] https://github.com/carbon-design-system/carbon-website/issues/3849
- [ ] https://github.com/carbon-design-system/carbon-website/issues/3847
- [ ] https://github.com/carbon-design-system/carbon-website/issues/3850
- [ ] https://github.com/carbon-design-system/carbon-website/issues/3851
- [ ] https://github.com/carbon-design-system/carbon/issues/15370

**Comment by tay1orjones** [comment]
When thinking about how to surface the status of these 4 categories on the website, we may be able to generate a json file that could be consumed by the website to dynamically power the table (or however it's designed) on the website.

This could include:
- As part of a release in CI, generate the JSON report using [playwright's JSON reporter](https://playwright.dev/docs/test-reporters#json-reporter). This file contains all the information about the tests, status, result, etc.
- Add this JSON file to the website, and build a component to pull in the file and generate the table or other design for these statuses.
  - This component could also use the github API to pull in the number of issues related to a given component and surface that alongside the status (if we want to)
- In the monorepo, the tests for each component could follow a naming convention to build this out, e.g. `Button - default state`, `Button - complex state`, `Button - keyboard navigation`. These names would be used by the component in the website to determine if the a11y statuses 1, 2, 3, 4 should be marked as ✅ or 🚫 
- That same release action or CI workflow in the monorepo could open a PR on the website repo updating the JSON file with the newest version as part of the release, so this would automatically be kept up to date.
